packages/core/src/codewhisperer/util/authUtil.ts

```diff
@@ -299,17 +299,33 @@ export class AuthUtil {
             this.secondaryAuth.isConnectionExpired &&
             this.conn !== undefined &&
             isValidCodeWhispererCoreConnection(this.conn)
-        getLogger().info(`codewhisperer: Connection expired = ${connectionExpired},
-                           secondaryAuth connection expired = ${this.secondaryAuth.isConnectionExpired},
-                           connection is undefined = ${this.conn === undefined}`)
-        if (this.conn) {
-            getLogger().info(
-                `codewhisperer: isValidCodeWhispererConnection = ${isValidCodeWhispererCoreConnection(this.conn)}`
-            )
-        }
+
+        this.logCnnection()
+
         return connectionExpired
     }
 
+    private logCnnection() {
+        const connectionValid = this.conn !== undefined && !this.secondaryAuth.isConnectionExpired
+        const connectionExpired =
+            this.secondaryAuth.isConnectionExpired &&
+            this.conn !== undefined &&
+            isValidCodeWhispererCoreConnection(this.conn)
```

Comment:
could avoid repeating this logic if `isConnectionExpired` and `isConnectionValid` took a `log:boolean` parameter:
```suggestion
        const connectionValid = isConnectionValid(false)
        const connectionExpired = isConnectionExpired(false)
```

Reply:
gotcha